### PR TITLE
Bug 1690253 - Implement the 'events' ping

### PR DESCRIPTION
**Pull request by brizental**
### Pull Request checklist ###

- [x] **Quality**: Make sure this PR builds and runs cleanly.
  - Inside the `glean/` folder, run:
    - `npm run test` Runs _all_ tests
    - `npm run lint` Runs _all_ linters
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry to `CHANGELOG.md` or an explanation of why it does not need one
- [x] **Documentation**: This PR includes documentation changes, an explanation of why it does not need that or a follow-up bug has been filed to do that work
   - https://github.com/mozilla/glean/pull/1988

### CODE REVIEW

File: glean/src/pings.yaml [hunk 1]
```diff
@@ -23,3 +23,36 @@ deletion-request:
...
+    - https://bugzilla.mozilla.org/1512938
+  data_reviews:
+    - https://bugzilla.mozilla.org/show_bug.cgi?id=1512938#c3
+  notification_emails:
+    - glean-team@mozilla.com
+  reasons:
+    startup: |
+      The ping was submitted at startup.
+      The events ping is always sent if there are any pending events at startup,
+      because event timestamps are not as reliable across application runs.
```
Comment (brizental) [inline glean/src/pings.yaml:48]: We did not discuss in the bug, but I though it made sense to also include the startup trigger for sending the events ping as well as the max_capacity.

File: glean/src/core/config.ts [hunk 2]
```diff
@@ -11,6 +11,10 @@ import { Context } from "./context.js";
 
 const LOG_TAG = "core.Config";
 
+// The default maximum amount of events Glean will store before submitting the events ping.
+// If the maximum is hit, the events ping is sent immediatelly.
```
Comment (badboy) [inline glean/src/core/config.ts:15]:
```suggestion
// If the maximum is hit, the events ping is sent immediately.
```

File: glean/src/core/config.ts [hunk 3]
```diff
@@ -35,6 +39,8 @@ export interface ConfigurationInterface {
   readonly appDisplayVersion?: string,
   // The server pings are sent to.
   readonly serverEndpoint?: string,
+  // The build date, provided by glean_parser
```
Comment (badboy) [inline glean/src/core/config.ts:42]:
```suggestion
  // The maximum number of events to store before submitting the events ping.
```

File: glean/src/core/config.ts [hunk 4]
```diff
@@ -11,6 +11,10 @@ import { Context } from "./context.js";
 
 const LOG_TAG = "core.Config";
 
+// The default maximum amount of events Glean will store before submitting the events ping.
```
Comment (badboy) [inline glean/src/core/config.ts:14]:
```suggestion
// The default maximum number of events Glean will store before submitting the events ping.
```

File: glean/src/core/metrics/events_database/index.ts [hunk 5]
```diff
@@ -47,12 +48,14 @@ function createDateObject(str?: ExtraValues): Date {
  * Creates an execution counter metric.
  *
  * @param sendInPings The list of pings this metric is sent in.
+ *        Note: The 'events' ping should not contain glean.restarted events,
+ *        so this ping will be filtered out from the the 'sendInPings' array.
```
Comment (badboy) [inline glean/src/core/metrics/events_database/index.ts:52]:
```suggestion
 *        so this ping will be filtered out from the 'sendInPings' array.
```

File: glean/src/core/metrics/events_database/index.ts [hunk 6]
```diff
@@ -62,13 +65,15 @@ function getExecutionCounterMetric(sendInPings: string[]): CounterMetricType {
  * Creates an `glean.restarted` event metric.
  *
  * @param sendInPings The list of pings this metric is sent in.
+ *        Note: The 'events' ping should not contain glean.restarted events,
+ *        so this ping will be filtered out from the the 'sendInPings' array.
```
Comment (badboy) [inline glean/src/core/metrics/events_database/index.ts:69]:
```suggestion
 *        so this ping will be filtered out from the 'sendInPings' array.
```

File: glean/src/core/upload/manager.ts [hunk 7]
```diff
@@ -35,6 +35,10 @@ export interface QueuedPing extends PingInternalRepresentation {
 class PingUploadManager implements PingsDatabaseObserver {
   // A FIFO queue storing a `QueuedPing` for each pending ping.
   private queue: QueuedPing[];
+  // A set of the idenfitifers of pings being processed
```
Comment (badboy) [inline glean/src/core/upload/manager.ts:38]:
```suggestion
  // A set of the identifiers of pings being processed
```

**Review by badboy — APPROVED** [review]

**Pull request merged**